aptos-move/framework/aptos-framework/sources/vesting.move

```diff
@@ -1460,6 +1475,84 @@ module aptos_framework::vesting {
+        admin: &signer,
+        operator: &signer,
+    ) acquires AdminStore, VestingContract {
+        let admin_address = signer::address_of(admin);
+        let operator_address = signer::address_of(operator);
+        setup(aptos_framework, &vector[admin_address, @11, operator_address]);
+        let contract_address = setup_vesting_contract(
+            admin, &vector[@11], &vector[GRANT_AMOUNT], admin_address, 10);
+
+        update_operator_with_same_commission(admin, contract_address, operator_address);
+        assert!(operator(contract_address) == operator_address, 0);
+        assert!(operator_commission_percentage(contract_address) == 10, 0);
+
+        let new_commission_percentage = 7;
+        update_commission_percentage(admin, contract_address, new_commission_percentage);
+        assert!(operator_commission_percentage(contract_address) == new_commission_percentage, 0);
+    }
+
+    #[test(aptos_framework = @0x1, admin = @0x123, shareholder = @0x234, operator = @0x345)]
+    public entry fun test_commission_percentage_change(
+        aptos_framework: &signer,
+        admin: &signer,
+        shareholder: &signer,
+        operator: &signer,
+    ) acquires AdminStore, VestingContract {
+        let admin_address = signer::address_of(admin);
+        let operator_address = signer::address_of(operator);
+        let shareholder_address = signer::address_of(shareholder);
+        setup(aptos_framework, &vector[admin_address, shareholder_address, operator_address]);
+        let contract_address = setup_vesting_contract(
+            admin, &vector[shareholder_address], &vector[GRANT_AMOUNT], admin_address, 0);
+        assert!(operator_commission_percentage(contract_address) == 0, 0);
+        let stake_pool_address = stake_pool_address(contract_address);
+
+        // 10% commission will be paid to the operator.
+        update_operator(admin, contract_address, operator_address, 10);
+
+        // Operator needs to join the validator set for the stake pool to earn rewards.
+        let (_sk, pk, pop) = stake::generate_identity();
+        stake::join_validator_set_for_test(&pk, &pop, operator, stake_pool_address, true);
+        stake::assert_stake_pool(stake_pool_address, GRANT_AMOUNT, 0, 0, 0);
+        assert!(get_accumulated_rewards(contract_address) == 0, 0);
+        assert!(remaining_grant(contract_address) == GRANT_AMOUNT, 0);
+
+        let expected_commission = 0;
+
+        // Stake pool earns some rewards.
+        stake::end_epoch();
+        let (_, accumulated_rewards, _) = staking_contract::staking_contract_amounts(contract_address, operator_address);
+
+        // Update commission percentage to 20%. This also immediately requests commission.
+        update_commission_percentage(admin, contract_address, 20);
+        // Commission is calculated using the previous commission percentage which is 10%.
+        expected_commission = expected_commission + (accumulated_rewards / 10);
```

Comment:
Nit: why not just define expected_commission here since it’s 0 before here anyway

Reply:
Updated accordingly.